include/turtle/detail/verifiable.hpp

```diff
@@ -25,6 +25,8 @@ namespace detail
         virtual bool verify() const = 0;
 
         virtual void reset() = 0;
+
+        virtual void context_destroyed() = 0;
```

Comment:
Not overly fond of the name of this function, can't we find something more in line with the rest of the interface here?

Reply:
Ideas? I'm not fond of it either. Better would be to initialize the context member in the ctor which forces correct order. But that would be more changes.